apps/rush-lib/src/cli/actions/PurgeAction.ts

```diff
@@ -34,22 +34,21 @@ export class PurgeAction extends BaseRushAction {
   }
 
   protected run(): Promise<void> {
-    return Promise.resolve().then(() => {
```

Comment:
>return Promise.resolve().then(() => { [](start = 4, length = 37)

We chatted about this, and there are two possible approaches we could take in our code base (that really need to be consistent everywhere):

1. Every function that returns a Promise must guarantee that it doesn't throw any synchronous exceptions, e.g. using the wrapper shown here.

2. Every root caller that starts a promise chain has responsibility for catching synchronous exceptions.

Because we can't think of a way to implement # 2 that would work with `Promise.all(f(x,y), g())`, it seems like # 1 is the only viable option.  If so let's try to apply that everywhere in the Rush code base.